frontend/packages/console-telemetry-plugin/src/listeners/segment.ts

```diff
@@ -110,12 +110,18 @@ export const eventListener: TelemetryEventListener = async (
   eventType: string,
   properties?: any,
 ) => {
-  if (TELEMETRY_DEBUG) {
+  if (TELEMETRY_DISABLED || !apiKey) {
```

Comment:
I don't want to add further changes to this PR since it's already got an lgtm, but I might follow on with a tweak like this:
```suggestion
  if (!apiKey) {
    // eslint-disable-next-line no-console
    console.debug(
      'console-telemetry-plugin: segment API key missing - ignoring telemetry event:',
      eventType,
      properties,
    );
    return;
```